### PR TITLE
Translate documentation for Navigation, both Pattern and Guideline

### PR DESCRIPTION
**Pull request by zachguo**
According to comments [here](https://github.com/ant-design/ant-design/issues/1471#issuecomment-313475226) and [here](https://github.com/ant-design/ant-design/issues/1471#issuecomment-300466860), ping reviewers @choudlet @abenhamdine . Please take a look.

### CODE REVIEW

**Comment by abenhamdine** [comment]
Nice work !
I'm not a native english-speaker, but it reads pretty well.

**Comment by afc163** [comment]
@choudlet @abenhamdine

File: docs/pattern/navigation.en-US.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+order: 1
+title:
```
Comment (pierreneter) [inline docs/pattern/navigation.en-US.md]: Just keep the English title

File: docs/spec/navigation.en-US.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+---
+order: 8
+title:
```
Comment (pierreneter) [inline docs/spec/navigation.en-US.md]: And here

**Review by pierreneter — CHANGES_REQUESTED** [review]

**Comment by pierreneter** [comment]
Related to #6931

**Comment by zachguo** [comment]
@pierreneter fixed and squashed

File: docs/pattern/navigation.en-US.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+---
+order: 1
+title: Navigation
+---
+
+Broadly speaking, anything telling user where to go and how to get there can be called navigation. We abstract common navigation patterns and provide handy components to helper designers and developers to build a clear and smooth navigational system. When using navigation or customizing navigational structures, please pay attention to following common pitfalls:
```
Comment (benjycui) [inline docs/pattern/navigation.en-US.md]: `任何告知用户他在哪里` is missing.

**Comment by benjycui** [comment]
👍

File: docs/pattern/navigation.en-US.md (hunk 3 above)
Comment (yesmeck) [inline docs/pattern/navigation.en-US.md]: helper => help

Comment (zachguo) [inline docs/pattern/navigation.en-US.md]: 这里的“When using navigation or customizing navigational structures”根据原文翻的，但感觉很chinglish，直接简写成“When using and customizing navigation”如何？

**Review by abenhamdine — APPROVED** [review]

**Review by pierreneter — APPROVED** [review]

**Pull request merged**